metals/src/main/scala/scala/meta/internal/metals/Diagnostics.scala

```diff
@@ -306,4 +308,30 @@ final class Diagnostics(
     toPublish
   }
 
+  private def adjustedDiagnosticData(
+      diagnostic: l.Diagnostic,
+      edit: TokenEditDistance,
+  ): Option[Object] =
+    diagnostic match {
+      case ScalacDiagnostic.ScalaDiagnostic(Left(textEdit)) =>
+        edit
+          .toRevised(
+            textEdit,
+            shouldAdjustWithinToken(diagnostic),
+            fallbackToNearest = false,
+          )
+          .map(_.toJsonObject)
```

Comment:
I found setting the data to empty when it was out of date a bit trickier than expected, since `toRevised` for ranges only returns none when the source doesn't tokenize.  I added a param `fallbackToNearest` to turn this off and so return `None` also when there's no match.